akka-stream/src/main/scala/akka/stream/scaladsl/Framing.scala

```diff
@@ -266,6 +266,8 @@ object Framing {
           frameSize = parsedLength + minimumChunkSize
           if (frameSize > maximumFrameLength) {
             failStage(new FramingException(s"Maximum allowed frame size is $maximumFrameLength but decoded frame header reported size $frameSize"))
+          } else if (frameSize < 1) {
```

Comment:
Is it arguable that a length field of zero is valid? E.g. a client sending some form of keep-alive where there is no actual data, so length is set to zero. This isn't an issue for our use case but I wonder if others may be sending such packets in which case they will get failures when they upgrade. Instead, the current behaviour (of sending an empty `ByteString`) would be maintained if this specified `frameSize < 0`.